src/werkzeug/local.py

```diff
@@ -46,41 +89,59 @@ def release_local(local: Union["LocalStack", "Local"]) -> None:
 
 
 class Local:
-    __slots__ = ("__storage__", "__ident_func__")
+    __slots__ = ("_storage", "_ident_func")
 
     def __init__(self) -> None:
-        object.__setattr__(self, "__storage__", {})
-        object.__setattr__(self, "__ident_func__", get_ident)
+        object.__setattr__(self, "_storage", ContextVar("local_storage"))
+        object.__setattr__(self, "_ident_func", get_ident)
+
+    @property
+    def __storage__(self):
+        return self._storage.get({})
```

Comment:
Seems that this isn't used anymore, so it should show a deprecation warning.

Reply:
Done